servers/src/main/java/tachyon/worker/block/BlockHeartbeatReporter.java

```diff
@@ -130,9 +132,17 @@ private void addBlockToAddedBlocks(long blockId, long storageDirId) {
    * @param blockId The block to remove
    */
   private void removeBlockFromAddedBlocks(long blockId) {
-    for (List<Long> blockList : mAddedBlocks.values()) {
+    Iterator<Entry<Long, List<Long>>> iterator = mAddedBlocks.entrySet().iterator();
```

Comment:
We have to use `iterators` instead of more friendly `for-each` here for the reason that we need to modify collection in the loop.